File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3

import rclpy 
from rclpy.node import Node
from example_interfaces.msg import String


class SmartphoneNode (Node) :
    def __init__(self) :
        super().__init__("smartphone")
        # Creating the subscriber 
        # Takes four params data type and topic name a callback and a queue size
        self.subcsubscriber_ = self.create_subscription(String, "robot_news", self.callback_robot_news, 10)
        self.get_logger().info("smartphone has been started.")
       
    def callback_robot_news (self, msg : String) :
        self.get_logger().info(msg.data)
    


    





def main (args=None) :
    
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__" :
    main()

File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"



class Smarthphone : public rclcpp::Node{

    public:
    Smarthphone () : Node("smartphone") {
       
        subscriber_ = this->create_subscription<example_interfaces::msg::String>("robot_news",10, [this] (const example_interfaces::msg::String::SharedPtr msg) {call_back_robot_news(msg);});
        RCLCPP_INFO(this->get_logger(), "Smartphone has been started.");

    };

    private:
    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;

    void call_back_robot_news (const example_interfaces::msg::String::SharedPtr msg) {
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }
    
    

};


int main (int argc, char **argv) {

    
    rclcpp::init(argc, argv);    
    auto node {std::make_shared<Smarthphone>()};
    rclcpp::spin(node);
    rclcpp::shutdown();


    return 0;

}